src/Ast.ml

```diff
@@ -261,26 +266,23 @@ module Structure_item : Module_item with type t = structure_item = struct
       | Pstr_open _ -> Location.is_single_line itm.pstr_loc c.Conf.margin
       | _ -> false )
 
-  let allow_adjacent (itmI, cI) (itmJ, cJ) =
-    match Conf.(cI.module_item_spacing, cJ.module_item_spacing) with
-    | `Compact, `Compact -> (
-      match (itmI.pstr_desc, itmJ.pstr_desc) with
-      | Pstr_eval _, Pstr_eval _
-       |Pstr_value _, Pstr_value _
-       |Pstr_primitive _, Pstr_primitive _
-       |(Pstr_type _ | Pstr_typext _), (Pstr_type _ | Pstr_typext _)
-       |Pstr_exception _, Pstr_exception _
-       |( (Pstr_module _ | Pstr_recmodule _ | Pstr_open _ | Pstr_include _)
-        , (Pstr_module _ | Pstr_recmodule _ | Pstr_open _ | Pstr_include _)
-        )
-       |Pstr_modtype _, Pstr_modtype _
-       |Pstr_class _, Pstr_class _
-       |Pstr_class_type _, Pstr_class_type _
-       |Pstr_attribute _, Pstr_attribute _
-       |Pstr_extension _, Pstr_extension _ ->
-          true
-      | _ -> false )
-    | _ -> true
+  let allow_adjacent itmI cI itmJ cJ =
```

Comment:
Pairing the item and the config that pertains to it is intended to help clarify the meaning of the arguments. Also, elsewhere the common idiom is to use such pairs. Were they removed here for any particular reason?